plugins/signals-node/api-report.md

```diff
@@ -0,0 +1,57 @@
+## API Report File for "@backstage/plugin-signals-node"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+/// <reference types="node" />
+
+import { Duplex } from 'stream';
+import { EventBroker } from '@backstage/plugin-events-node';
+import { EventParams } from '@backstage/plugin-events-node';
+import { EventSubscriber } from '@backstage/plugin-events-node';
+import { IdentityApi } from '@backstage/plugin-auth-node';
+import { IncomingMessage } from 'http';
+import { JsonObject } from '@backstage/types';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { ServiceRef } from '@backstage/backend-plugin-api';
+
+// @public (undocumented)
+export type ServiceOptions = {
+  eventBroker?: EventBroker;
+  logger: LoggerService;
+  identity: IdentityApi;
+};
+
+// @public (undocumented)
+export type SignalEventBrokerPayload = {
+  recipients?: string[];
+  topic?: string;
+  message?: JsonObject;
+};
+
+// @public (undocumented)
+export class SignalService implements EventSubscriber {
```

Comment:
Could you restructure the implementation so that the `SignalService` implementation doesn't directly implement `EventSubscriber`, but instead keeps the `EventBroker` subscriber internal?

We also need to split out the implementation and make this an interface that only lists the public methods. Something like `SignalService` as the interface and `DefaultSignalService` as the implementaiton.